presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1077,6 +1082,10 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
 
         List<PartitionUpdate> partitionUpdates = getPartitionUpdates(fragments);
 
+        if (isWritingStagingFilesEnabled(session)) {
+            stagingFileCommitter.commitFiles(session, handle.getSchemaName(), handle.getTableName(), partitionUpdates);
+        }
```

Comment:
add state check when this not enabled, and staging files should be same as target files.